Добавлены файлы программы для оценки марком

Ivanstarterprog opened the pull request.

Ivanstarterprog committed:
Добавлены файлы программы

Hamjoshua commented:
Одобрено

Silmarill commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,435 @@
+                int thisMatrixWeight, comparedMatrixWeight;
+                thisMatrixWeight = 0;
+                comparedMatrixWeight = 0;
+                for (int rowIndex = 0; rowIndex < sideSize; ++rowIndex)
```
лучше отделять смысловые блоки пустой строкой для повышения читаемости

Silmarill commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,435 @@
+    {
+        static void Main(string[] args)
+        {
+            SquareMatrix test = new SquareMatrix(4);
```
Я разочарован именем переменной.

Silmarill commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,435 @@
+        {
+            SquareMatrix test = new SquareMatrix(4);
+            Console.WriteLine(test);
+            SquareMatrix test2 = new SquareMatrix(4);
```
Я разочарован именем переменной 2.

Silmarill commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,435 @@
+            Console.WriteLine(test.GetHashCode());
+            Console.WriteLine(test.GetDeterminant(test));
+            Console.WriteLine(test.ReverseMatrix());
+            SquareMatrix test3 = new SquareMatrix(5);
```
Я разочарован именем переменной 3

Silmarill commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,435 @@
+        public object Clone()
+        {
+            SquareMatrix clonedMatrix = new SquareMatrix(sideSize);
+            clonedMatrix.matrix = matrix;
```
Нужно обоснование, что копирование здесь - глубокое.

Silmarill commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,435 @@
+
+    internal class Program
+    {
+        static void Main(string[] args)
```
Не увидел само клонирование.

Silmarill reviewed (changes requested):
Need corrections.

Ivanstarterprog committed:
Исправлены ошибки в именах переменных (Разработчик получил линейкой по рукам за переменные test)
Добавлен блок кода, подтверждающий что копирование глубокое. И что оно в целом происходит

Hamjoshua commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,443 @@
+    }
+    class SquareMatrix : IComparable, ICloneable
+    {
+        private int sideSize { get; set; }
```
Приватные поля и свойства должны оформляться так: _camelCase

Hamjoshua commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,443 @@
+    {
+        private int sideSize { get; set; }
+        private int[,] matrix;
+        static Random RandomInt = new Random();
```
Статические приватные поля должны оформляться так: st_camelCase

Hamjoshua commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,443 @@
+            return matrixStringBuilder.ToString();
+        }
+        
+        public object Clone()
```
Это копирование не глубокое

Hamjoshua commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,443 @@
+                for (int columnIndex = 0; columnIndex < squareMatrix.sideSize; ++columnIndex)
+                {
+                    int minor = Convert.ToInt32(Math.Pow(-1, columnIndex));
+                    int ColumnNumber = minor * squareMatrix.matrix[0, columnIndex];
```
Все переменные должны быть либо с большой буквы, либо с маленькой. В твоем коде доминируют camelCase

Hamjoshua replied:
Следовательно, все переменные должны быть в camelCase

Hamjoshua commented on SquareMatrix/Program.cs:
```diff
@@ -0,0 +1,443 @@
+                SquareMatrix comparedMatrix = other as SquareMatrix;
+                int thisMatrixWeight, comparedMatrixWeight;
+                thisMatrixWeight = 0;
+                comparedMatrixWeight = 0;
```
Разделяй смысловые блоки Ентером для улучшения читаемости

Hamjoshua reviewed (changes requested):
Отправлено на переработку

Ivanstarterprog committed:
Исправленны имена переменных
Добавлено глубокое клонирование

Ivanstarterprog committed:
Добавлена перегрузка оператора приведения в String
